mina.netty/src/main/java/org/jboss/netty/channel/socket/nio/AbstractNioSelector.java

```diff
@@ -351,8 +355,9 @@ public void run() {
                     shutdownLatch.countDown();
                     break;
                 } else {
-                    process(selector);
-                    processRead();
+                    workCount += process(selector);
+                    workCount += processRead();
+                    idleStrategy.idle(workCount);
```

Comment:
The intent might be clearer if the `idleStrategy.idle(workCount)` is moved immediately outside this `else` clause, since `workCount` is also modified before the `else` clause. The main reason this all works is due to the `break` at line 356 above, but that's a bit too subtle and moving `idleStrategy.idle(workCount)` is probably more clear.